### PR TITLE
msmtpq: fix errors when `$Q` contains spaces

### LINKED ISSUE #87: msmtpq doesn’t work properly if `$Q` contains spaces
## Steps to reproduce

1. `mkdir -p ~/Spaces\ in\ path/Q`
2. `export Q=~/Spaces\ in\ path/Q`
3. Disconnect from the Internet (or do something else that will make `msmtpq` fail to send a message).
4. `printf 'Subject: Test subject\n\nTest Body.\n' | msmtpq <some-email-address>`
5. `msmtpq --q-mgmt -d`

That will tell you that there’s no messages in the queue, even though there’s supposed to be one in the queue.

### PR DESCRIPTION
**Pull request by Jayman2000**
Fixes #87.

### CODE REVIEW

File: scripts/msmtpq/msmtpq [hunk 1]
```diff
@@ -308,38 +308,49 @@ send_queued_mail() {   # <-- mail id
...
-      CNT=$((CNT + 1))
-      dsp '' "mail  num=[ $CNT ]  id=[ $ID ]"                  # show mail id ## patch in
-      'grep' -E -s --colour -h -m 3 '(^From:|^To:|^Subject:)' "$M" # show mail info
-      [ -n "$2" ] && Q_LST[$CNT]="$ID" # bang mail id into array (note 1-based array indexing)
-    done
-    echo
-  else                               # no mails ; no contents
+  for M in "$Q"/*.mail ; do          # cycle through each
+    ID="$(basename "$M" .mail)"      # take mail id from filename
+    CNT=$((CNT + 1))
```
Comment (Jayman2000) [inline scripts/msmtpq/msmtpq:347]: I should probably quote `$((CNT + 1))`. It’s not necessary, but I want to set a precedent: quote your variables unless there’s a reason not to.

File: scripts/msmtpq/msmtpq [hunk 2]
```diff
@@ -308,38 +308,49 @@ send_queued_mail() {   # <-- mail id
...
+    dsp '' "mail  num=[ $CNT ]  id=[ $ID ]"                  # show mail id ## patch in
+    'grep' -E -s --colour -h -m 3 '(^From:|^To:|^Subject:)' "$M" # show mail info
+    [ -n "$2" ] && Q_LST[$CNT]="$ID" # bang mail id into array (note 1-based array indexing)
```
Comment (Jayman2000) [inline scripts/msmtpq/msmtpq:350]: I should probably quote `$CNT` here.

**Review by Jayman2000 — COMMENTED** [review]
I just noticed some things that I want to change before this gets accepted.

File: scripts/msmtpq/msmtpq (hunk 1 above)
Comment (Jayman2000) [reply scripts/msmtpq/msmtpq]: Fixed in a force push.

File: scripts/msmtpq/msmtpq (hunk 2 above)
Comment (Jayman2000) [reply scripts/msmtpq/msmtpq]: Fixed in a force push.

**Comment by marlam** [comment]
This is now included. Thanks a lot!

**Pull request closed without merging**